modules/python/python-dest.c

```diff
@@ -86,11 +86,12 @@ python_dd_set_class(LogDriver *d, gchar *class_name)
 }
 
 void
-python_dd_set_option(LogDriver *d, gchar *key, gchar *value)
+python_dd_add_option(LogDriver *d, const gchar *name, PyObject *value)
```

Comment:
Can't we encapsulate the hashtable and related functions into a PythonOptions (or similarly named) structure which the python components would just contain?

Reply:
My first implementation was like you have mentioned, but half way through it, I felt like it had too much boilerplate. However, I finished that implementation now and it is okay, I guess. Now we have 2 implementations, so we can decide :)